lib/IRGen/GenStruct.cpp

```diff
@@ -956,24 +1092,21 @@ class ClangRecordLowering {
         continue;
       }
 
-      VarDecl *swiftField;
-      if (sfi != sfe) {
-        swiftField = *sfi;
-        if (isImportOfClangField(swiftField, clangField)) {
-          ++sfi;
-        } else {
-          swiftField = nullptr;
+      VarDecl *swiftField = nullptr;
+      for (auto *sfi : swiftProperties) {
+        if (sfi->getClangDecl() == clangField) {
```

Comment:
Can we not rely on `getStoredProperties()` being in order?

Reply:
Not anymore. We discussed this offline a couple of weeks ago. This is part of the lazy member loading patch.

Reply:
This isn't a technical limitation of how lazy member loading is implemented. It's just how it happened to end up. This makes some other code simpler, but I can preserve the order if it's important for some reason.